routing/base/astar_algorithm.hpp

```diff
@@ -343,40 +357,48 @@ typename AStarAlgorithm<TGraph>::Result AStarAlgorithm<TGraph>::FindPath(
     return true;
   };
 
-  auto adjustEdgeWeight = [&](TVertexType const & vertex, TEdgeType const & edge) {
+  auto adjustEdgeWeight = [&](State const & stateV, TEdgeType const & edge, TWeightType & weight) {
     auto const len = edge.GetWeight();
-    auto const piV = graph.HeuristicCostEstimate(vertex, finalVertex);
+    auto const piV = graph.HeuristicCostEstimate(stateV.vertex, finalVertex);
     auto const piW = graph.HeuristicCostEstimate(edge.GetTarget(), finalVertex);
     auto const reducedLen = len + piW - piV;
 
     CHECK(reducedLen >= -kEpsilon, ("Invariant violated:", reducedLen, "<", -kEpsilon));
-    return std::max(reducedLen, kZeroDistance);
+
+    auto const fullLen = len + stateV.distance + piS - piV;
+    if (!checkWeightCallback(fullLen))
```

Comment:
checkLengthCallback?

Reply:
`std::function<bool(TWeightType const &)>` looks like CheckWeightCallback to me :)